torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -760,12 +760,15 @@ def _need_prefetch_post_backward_hook(self) -> bool:
         else:
             return False
 
+    @staticmethod
     @contextlib.contextmanager
-    def state_dict_type(self, state_dict_type: StateDictType) -> Generator:
+    def state_dict_type(module: nn.Module, state_dict_type: StateDictType) -> Generator:
```

Comment:
maybe rename it to `fsdp_state_dict_type`, unless you feel that the call `FSDP.state_dict_type()` is already clear enough that it's setting things for only FSDP modules.

Reply:
I feel `FSDP.state_dict_type()` explicit enough. Only FSDP internal code will use something like `module.state_dict_type()` or `self.state_dict_type()`.